src/batch/src/task/task_execution.rs

```diff
@@ -570,6 +581,13 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
         };
     }
 
+    pub fn cancel(&self) {
+        if let Some(sender) = self.shutdown_tx.lock().take() {
+            // Drop sender directly to mark cancel without error.
+            drop(sender);
+        };
+    }
```

Comment:
Maybe it's better here to explicitly send a cancel message instead of directly dropping the sender. Relying on the `drop` of a variable doesn't seem robust. 🤔